trin-core/src/types/block_body.rs

```diff
@@ -178,6 +180,42 @@ impl Transaction {
     }
 }
 
+impl<'de> Deserialize<'de> for Transaction {
```

Comment:
Why do we want to only deserialize a transaction? I think it is good practice to implement both Serialize and Deserialize traits.